xcell/cls/theory.py

```diff
@@ -119,10 +119,49 @@ def compute_tracer_ccl(self, name, tracer, mapper):
                 'with_hm': with_hm,
                 'normed': normed_profile}
 
+    def get_ccl_tkka(self, ccl_trA1, ccl_trA2, ccl_trB1, ccl_trB2,
+                     kind='1h'):
+        if kind not in ['1h']:
+            raise NotImplementedError(f"Non-Gaussian term {kind} "
+                                      "not supported.")
+
+        pA1 = ccl_trA1['ccl_pr']
+        pA2 = ccl_trA2['ccl_pr']
+        if ccl_trA1['name'] == ccl_trA2['name']:
+            pr2ptA = ccl_trA1['ccl_pr_2pt']
+            pA2 = pA1
+        else:
+            pr2ptA = None
+        pB1 = ccl_trB1['ccl_pr']
+        pB2 = ccl_trB2['ccl_pr']
+        if ccl_trB1['name'] == ccl_trB2['name']:
+            pr2ptB = ccl_trB1['ccl_pr_2pt']
+            pB2 = pB1
+        else:
+            pr2ptB = None
+
+        k_s = np.geomspace(1E-4, 1E2, 512)
+        lk_s = np.log(k_s)
+        a_s = 1./(1+np.linspace(0., 6., 30)[::-1])
+        tkk = ccl.halos.halomod_Tk3D_1h(self.cosmo, self.hm_par['calculator'],
+                                        prof1=pA1, prof2=pA2,
+                                        prof12_2pt=pr2ptA,
+                                        prof3=pB1, prof4=pB2,
+                                        prof34_2pt=pr2ptB,
+                                        normprof1=ccl_trA1['normed'],
+                                        normprof2=ccl_trA2['normed'],
+                                        normprof3=ccl_trB1['normed'],
+                                        normprof4=ccl_trB2['normed'],
+                                        a_arr=a_s, lk_arr=lk_s)
+        return tkk
+
     def get_ccl_pk(self, ccl_tr1, ccl_tr2):
         if ccl_tr1['with_hm'] or ccl_tr2['with_hm']:
+            p1 = ccl_tr1['ccl_pr']
+            p2 = ccl_tr2['ccl_pr']
             if ccl_tr1['name'] == ccl_tr2['name']:
                 pr2pt = ccl_tr1['ccl_pr_2pt']
+                p2 = p1
```

Comment:
hmmm... if they're the same tracer, it guess it is ok that `p2` is exactly the same dictionary as `p1` (it points to the same memory location). However, could there be any case when `p2` might be later modified and, therefore, be different to `p1`? If so, use `p1.copy()`

Reply:
Note that `p1` and `p2` are not dictionaries, but `ccl.halos.HaloProfile` objects, and if both are not **exactly** the same, ccl will throw an error if you attempt to use a `Profile2ptHOD` object for the auto-correlation. So yes, they need to be exactly the same profile object (at the memory level), and `.copy()` won't work.